lib/models/mirror.js

```diff
@@ -38,13 +43,27 @@ Mirror.set('toObject', {
 
 Mirror.statics.create = function(contract, contact, callback) {
   var self = this;
+
+  if (_.isEmpty(contract)) {
+    return callback(new Error('Cannot save mirror: invalid contract'));
+  }
+
+  if (_.isEmpty(contact)) {
+    return callback(new Error('Cannot save mirror: invalid contact'));
+  }
+
   var mirror = new self({
     shardHash: contract.data_hash,
     contact: contact.nodeID,
     contract: contract
   });
 
-  mirror.save(callback);
+  mirror.save(function(err, mirror) {
+    if (err) {
+      return callback(new Error('Cannot save mirror', err));
```

Comment:
Every other model that does a `.save()` follows this pattern. Is there a reason why `mirror.save` wouldn't?